myceli/src/listener.rs

```diff
@@ -178,9 +157,7 @@ impl Listener {
     }
 
     fn transmit_msg(&self, message: Message, target_addr: SocketAddr) -> Result<()> {
-        for chunk in self.chunker.chunk(message)? {
-            self.socket.send_to(&chunk, target_addr)?;
-        }
+        self.transport.send(message, &target_addr.to_string())?;
```

Comment:
I think you can just return a reference here if you receive a reference of the `target_addr`, but you'd have to change the API for the `send()` fn.

Reply:
Thanks for pointing this out! I was able to do a refactor around `sender_addr` which cleaned this up nicely.